packages/ui/src/components/va-button-group/VaButtonGroup.vue

```diff
@@ -1,74 +1,43 @@
     <va-config :components="buttonConfig">
       <slot />
     </va-config>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, computed, PropType } from 'vue'
+import { defineComponent, computed } from 'vue'
+import { extractComponentProps } from '../../utils/child-props'
 
-import { getGradientBackground } from '../../services/color-config/color-functions'
-import { useColors } from '../../composables/useColor'
-import { useTextColor } from '../../composables/useTextColor'
+import { useBem } from '../../composables/useBem'
+import { useDeprecatedProps } from '../../composables/useDeprecatedProps'
 import { useComponentPresetProp } from '../../composables/useComponentPreset'
 
 import { VaConfig } from '../va-config'
+import { VaButton } from '../va-button'
+
+const VaButtonProps = extractComponentProps(VaButton)
 
 export default defineComponent({
   name: 'VaButtonGroup',
   components: { VaConfig },
   props: {
+    ...VaButtonProps,
     ...useComponentPresetProp,
-    color: { type: String, default: 'primary' },
-    gradient: { type: Boolean, default: undefined },
-    textColor: { type: String, default: undefined },
-    rounded: { type: Boolean, default: true },
-    outline: { type: Boolean, default: false },
-    flat: { type: Boolean, default: false },
-    size: {
-      type: String as PropType<'medium' | 'small' | 'large'>,
-      default: 'medium',
-      validator: (value: string) => ['medium', 'small', 'large'].includes(value),
-    },
   },
-  setup (props) {
-    const { getColor } = useColors()
-    const colorComputed = computed(() => getColor(props.color))
-
-    const isTransparentBackground = computed(() => Boolean(props.outline || props.flat))
-    const { textColorComputed } = useTextColor(colorComputed, isTransparentBackground)
-
-    const computedBackground = computed(() => {
-      if (props.outline || props.flat) { return '' }
-
-      return props.gradient ? getGradientBackground(colorComputed.value) : colorComputed.value
-    })
-
-    const computedStyle = computed(() => {
-      const backgroundProperty = props.gradient ? 'background-image' : 'background'
-
-      return {
-        [backgroundProperty]: computedBackground.value,
-        color: textColorComputed.value,
-      }
-    })
 
-    const buttonConfig = computed(() => ({
-      VaButton: {
-        ...props,
-        color: props.gradient ? '#00000000' : props.color,
-        textColor: textColorComputed.value,
-      },
-    }))
-
-    const computedClass = computed(() => ({ 'va-button-group_square': !props.rounded }))
+  setup: (props) => {
+    // temp
+    useDeprecatedProps(['flat', 'outline'])
 
     return {
-      buttonConfig,
-      computedStyle,
-      computedClass,
+      buttonConfig: computed(() => ({ VaButton: { ...props } })),
+      computedClass: useBem('va-button-group', () => ({
+        square: !props.round,
+        small: props.size === 'small',
+        large: props.size === 'large',
+      })),
```

Comment:
Don't see why we need to pollute `return` with logic because it makes code harder to reuse.

Also we will have to move this away in case of future `<script setup>` usage.